pkg/util/ipset/ipset.go

```diff
@@ -72,74 +72,149 @@ var EntryMemberPattern = "(?m)^(.*\n)*Members:\n"
 // ipset version output is similar to "v6.10".
 var VersionPattern = "v[0-9]+\\.[0-9]+"
 
+// Option is variadic initializer for IPSet
+type Option func(set *IPSet)
+
+// NewIPSet returns a new IPSet which is an Interface to a set
+func NewIPSet(name string, opts ...Option) (*IPSet, error) {
+	ret := IPSet{
+		name: name,
+	}
+	setIPSetDefaults(&ret)
+	for _, opt := range opts {
+		opt(&ret)
+	}
+	if err := ret.validate(); err != nil {
+		return &IPSet{}, err
+	}
+	return &ret, nil
```

Comment:
Why not return the value instead of the pointer?  Such a small struct is unlikely to be a real problem, and pointers just make everything worse - unless theres a good reason to use pointer?